jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -512,33 +512,27 @@ private void checkAccess(Class<?> targetClass, String name, int memberModifiers,
 		 * @throws IllegalAccessException If the {@link Class} is not accessible from {@link #accessClass}. 
 		 */
 		private void checkClassAccess(Class<?> targetClass) throws IllegalAccessException {
-			/*[IF Sidecar19-SE]*/
-			checkClassModuleVisibility(accessMode, accessClass.getModule(), targetClass);
-			/*[ENDIF]*/
-			if (NO_ACCESS != accessMode) {
-				/* target class should always be accessible to the lookup class when they are the same class */
-				if (accessClass == targetClass) {
-					return;
-				}
-				
-				int modifiers = targetClass.getModifiers();
-				
-				/* A protected class (must be a member class) is compiled to a public class as
+			int targetModifiers = targetClass.getModifiers();
+			boolean accessible = false;
+			if (Modifier.isPublic(targetModifiers) || Modifier.isProtected(targetModifiers)) {
+				/*[IF Sidecar19-SE]*/
+				checkClassModuleVisibility(accessMode, accessClass.getModule(), targetClass);
+				/*[ENDIF]*/
+				accessible = true;
+			} else if (NO_ACCESS != accessMode) {
+				/* The target class should always be accessible to the lookup class when they are the same class.
+				 * A protected class (must be a member class) is compiled to a public class as
 				 * the protected flag of this class doesn't exist on the VM level (there is no 
-				 * access flag in the binary form representing 'protected')
+				 * access flag in the binary form representing 'protected').
 				 */
-				if (Modifier.isPublic(modifiers) || Modifier.isProtected(modifiers)) {
-					/* Already determined that we have more than "no access" (public access) */
-					return;
-				} else {
-					if (((PACKAGE == (accessMode & PACKAGE)) || Modifier.isPrivate(accessMode)) && isSamePackage(accessClass, targetClass)) {
-						return;
-					}
-				}
+				accessible = ((accessClass == targetClass)
+						|| ((PACKAGE == (accessMode & PACKAGE)) && isSamePackage(accessClass, targetClass))
```

Comment:
The code is clearer if this is written as a couple of if statements rather than a set of conditions assigned to the variable.

It also has that floating `);`

Reply:
Good plan.  Will fix.